macos/file_picker/Sources/file_picker/FilePickerPlugin.swift

```diff
@@ -2,181 +2,213 @@ import Cocoa
+                result(nil)
+                return
+            }
+            dialog.beginSheetModal(for: appWindow) { response in
+                // User dismissed the dialog
+                if (response != .OK) {
+                    result(nil)
+                    return
+                }
+                
+                if let url = dialog.url {
+                    result(url.path)
+                    return
+                }
+                
+                result(nil)
+            }
+        }
     }
-
-    let extensions = args["allowedExtensions"] as? [String] ?? []
-    applyExtensions(dialog, extensions)
-
-    guard let appWindow = getFlutterWindow() else {
-      result(nil)
-      return
+    
+    private func handleSaveFile(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        if checkEntitlement(requiredMode: .requireWrite, result: result) {
+            
+            let dialog = NSSavePanel()
+            let args = call.arguments as! [String: Any]
+            
+            dialog.title = args["dialogTitle"] as? String ?? ""
+            dialog.showsTagField = false
+            dialog.showsHiddenFiles = false
+            dialog.canCreateDirectories = true
+            dialog.nameFieldStringValue = args["fileName"] as? String ?? ""
+            
+            if let initialDirectory = args["initialDirectory"] as? String,
+               !initialDirectory.isEmpty
+            {
+                dialog.directoryURL = URL(fileURLWithPath: initialDirectory)
+            }
+            
+            let extensions = args["allowedExtensions"] as? [String] ?? []
+            applyExtensions(dialog, extensions)
+            
+            guard let appWindow = getFlutterWindow() else {
+                result(nil)
+                return
+            }
+            dialog.beginSheetModal(for: appWindow) { response in
+                // User dismissed the dialog
+                if (response != .OK) {
+                    result(nil)
+                    return
+                }
+                
+                if let url = dialog.url {
+                    result(url.path)
+                    return
+                }
+                
+                result(nil)
+            }
+        }
     }
-    dialog.beginSheetModal(for: appWindow) { response in
-      // User dismissed the dialog
-      if (response != .OK) {
-        result(nil)
-        return
-      }
-
-      if let url = dialog.url {
-        result(url.path)
-        return
-      }
-
-      result(nil)
+    
+    /// Checks if the  entitlements file contains the required entitlement for save files.
+    private func checkEntitlement(requiredMode: EntitlementMode, result: @escaping FlutterResult) -> Bool {
+        guard let task = SecTaskCreateFromSelf(nil) else {
+            result(FlutterError(code: "ENTITLEMENT_CHECK_FAILED", message: "Failed to create security task.", details: nil))
```

Comment:
Can we update the message here?

```suggestion
FlutterError(code: "ENTITLEMENT_CHECK_FAILED", message: "Failed to verify file_picker entitlements.", details: nil)
```

Reply:
Changed inside of a big commit